File: java_solutions/2151.maximum-good-people-based-on-statements.java
/*
 * @lc app=leetcode id=2151 lang=java
 *
 * [2151] Maximum Good People Based on Statements
 */

class Solution { public int maximumGood(int[][] A) { return IntStream.range(0, 1 << A.length).filter(mask -> IntStream.range(0, A.length).allMatch(i -> (mask & (1 << i)) == 0 || IntStream.range(0, A.length).allMatch(j -> A[i][j] == 2 || A[i][j] == ((mask >> j) & 1)))).map(mask -> Integer.bitCount(mask)).max().orElse(0); } }

File: kotlin_solutions_1/2151.maximum-good-people-based-on-statements.kt
/* @lc app=leetcode id=2151 lang=kotlin */
class Solution {
    fun maximumGood(A: Array<IntArray>) = (0 until (1 shl A.size)).filter { mask -> (0 until A.size).all { i -> (mask shr i and 1 == 0) || (0 until A.size).all { j -> A[i][j] == 2 || A[i][j] == (mask shr j and 1) } } }.maxOfOrNull { Integer.bitCount(it) } ?: 0
}